src/niscope/templates/session.py/fancy_fetch_array_measurement.py.mako

```diff
@@ -17,12 +17,10 @@
             self._fetch_num_records = -1 if num_records is None else num_records
             self._fetch_meas_num_samples = -1 if meas_num_samples is None else meas_num_samples
 
+        # For GrpcStubInterpreter, the server will automatically get _actual_meas_wfm_size, if needed.
```

Comment:
FYI, I wasn't sure if you can explicitly initialize fields to `None`, but mypy-protobuf's type annotations indicate that you can. (In addition to your testing.)

```
(.venv) PS C:\dev\nimi-python\generated\niscope> python.exe -m grpc_tools.protoc -I="..\..\src\niscope\metadata" -I="..\..\src\shared_protos" --python_out=. --grpc_python_out=. --mypy_out=. session.proto nidevice.proto niscope.proto
Writing mypy to session_pb2.pyi
Writing mypy to nidevice_pb2.pyi
Writing mypy to niscope_pb2.pyi
```

```
@typing_extensions.final
class FetchArrayMeasurementRequest(google.protobuf.message.Message):
    DESCRIPTOR: google.protobuf.descriptor.Descriptor

    VI_FIELD_NUMBER: builtins.int
    CHANNEL_LIST_FIELD_NUMBER: builtins.int
    TIMEOUT_FIELD_NUMBER: builtins.int
    ARRAY_MEAS_FUNCTION_FIELD_NUMBER: builtins.int
    ARRAY_MEAS_FUNCTION_RAW_FIELD_NUMBER: builtins.int
    MEAS_WFM_SIZE_FIELD_NUMBER: builtins.int
    @property
    def vi(self) -> session_pb2.Session: ...
    channel_list: builtins.str
    timeout: builtins.float
    array_meas_function: global___ArrayMeasurement.ValueType
    array_meas_function_raw: builtins.int
    meas_wfm_size: builtins.int
    def __init__(
        self,
        *,
        vi: session_pb2.Session | None = ...,
        channel_list: builtins.str = ...,
        timeout: builtins.float = ...,
        array_meas_function: global___ArrayMeasurement.ValueType = ...,
        array_meas_function_raw: builtins.int = ...,
        meas_wfm_size: builtins.int | None = ...,
    ) -> None: ...
```